test/unittests/testutil.h

```diff
@@ -523,14 +549,57 @@ class MockCatalogManager : public AbstractCatalogManager<MockCatalog> {
                                  const shash::Any  &catalog_hash,
                                  MockCatalog *parent_catalog);
   MockCatalog* RetrieveRootCatalog() { return GetRootCatalog(); }
+  void AddFile(const DirectoryEntryBase &entry,
+               const XattrList &xattrs,
+               const std::string &parent_directory)
+  {
+    ++num_added_files_;
+  }
+
   void RegisterNewCatalog(MockCatalog *new_catalog) {
     string mountpoint_str = new_catalog->root_path();
     PathString mountpoint(mountpoint_str.c_str(), mountpoint_str.length());
     catalog_map_[mountpoint] = new_catalog;
   }
 
- protected:
+  unsigned GetNumAutogeneratedCatalogs() { return autogenerated_catalogs_; }
+  unsigned GetNumAddedFiles() { return num_added_files_; }
+
+  MockCatalog *FindCatalog(const PathString &path) {
+    map<PathString, MockCatalog*>::iterator it;
+    string best_match = "";
+    unsigned best_match_length = 0;
+    for (it = catalog_map_.begin(); it != catalog_map_.end(); ++it) {
+      string key = it->first.ToString();
+      if (path.ToString().find(key) != string::npos) {
```

Comment:
Shouldn't the condition here be `path.ToString().find(key) == 0`?  In other words: `path` should start with `key`, no?  The current condition is true if `key` is found somewhere in `path`.